rfc/rfc-37/rfc-37.md

```diff
@@ -0,0 +1,286 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+# RFC-37: Metadata based Bloom Index
+
+## Proposers
+- @nsivabalan
+- @manojpec
+
+## Approvers
+ - @vinothchandar
+ - @satishkotha
+
+## Status
+JIRA: https://issues.apache.org/jira/browse/HUDI-2703
+
+## Abstract
+Hudi maintains several indices to locate/map incoming records to file groups during writes. Most commonly 
+used record index is the HoodieBloomIndex. Larger tables and global index has performance issues
+as the bloom filter from a large number of data files needed to be read and looked up. Reading from several
+files over the cloud object storage like S3 also faces request throttling issues. We are proposing to 
+build a new Metadata index (metadata table based bloom index) to boost the performance of existing bloom index. 
+
+## Background
+HoodieBloomIndex is used to find the location of incoming records during every write. Bloom index assists Hudi in
+deterministically routing records to a given file group and to distinguish inserts vs updates. This aggregate bloom
+index is built from several bloom filters stored in the base file footers. Prior to bloom filter lookup, the file
+pruning for the incoming records is also done based on the record key min/max stats stored in the base file footers.
+In this RFC, we plan to build a new index for the bloom filters under the metadata table which to assist in 
```

Comment:
Maybe we can call out that this overlaps heavily with Data Skipping part on the read-path (since for upserts we essentially running a query on PK) and hence we want to unify the infra underlying these execution paths

Reply:
Referred to RFC-27 DataSkiping Index.